connectors/src/connectors/google_drive/temporal/spreadsheets.ts

```diff
@@ -387,9 +387,24 @@ export async function syncSpreadSheet(
   );
 
   const sheetsAPI = google.sheets({ version: "v4", auth: oauth2client });
-  const spreadsheet = await sheetsAPI.spreadsheets.get({
-    spreadsheetId: file.id,
-  });
+  const spreadsheet = await (async () => {
+    try {
+      return await sheetsAPI.spreadsheets.get({
+        spreadsheetId: file.id,
+      });
+    } catch (err) {
+      if (isGAxiosServiceUnavailablError(err)) {
+        throw {
```

Comment:
Reading this code a second time, it feels weird to throw an object. Don't we have a `DustInternalError` class?